src/cli/src/main/java/org/locationtech/geogig/cli/plumbing/CreateIndex.java

```diff
@@ -10,51 +11,130 @@
+import org.locationtech.geogig.porcelain.LogOp;
+import org.locationtech.geogig.repository.Index;
+import org.locationtech.geogig.repository.Index.IndexType;
 import org.locationtech.geogig.repository.ProgressListener;
 import org.locationtech.geogig.repository.impl.GeoGIG;
+import org.locationtech.geogig.storage.IndexDatabase;
+import org.opengis.feature.type.GeometryType;
+import org.opengis.feature.type.PropertyDescriptor;
+import org.opengis.feature.type.PropertyType;
 
 import com.beust.jcommander.Parameter;
 import com.beust.jcommander.Parameters;
 import com.google.common.base.Optional;
-import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
 
 @RequiresRepository(true)
 @Parameters(commandNames = {
         "create-index" }, commandDescription = "Creates a spatial index for the specified feature tree")
 public class CreateIndex extends AbstractCommand implements CLICommand {
 
-    @Parameter(names = "--tree", required = true, description = "Name or path of the feature tree to create the index for")
-    private String treePath;
+    @Parameter(names = "--tree", required = true, description = "Name or path of the feature tree to create the index for.")
+    private String tree;
+
+    @Parameter(names = "--attribute", required = true, description = "Attribute to create the index for.")
+    private String attribute;
+
+    @Parameter(names = "--indexHistory", description = "If specified, indexes will be created for all commits in the history.")
+    private boolean indexHistory = false;
 
     @Override
     protected void runInternal(GeogigCLI cli)
             throws InvalidParameterException, CommandFailedException, IOException {
 
         GeoGIG geogig = cli.getGeogig();
-        CreateQuadTree command;
-        command = geogig.command(CreateQuadTree.class);
 
-        Optional<ObjectId> treeId = geogig.command(ResolveTreeish.class).setTreeish(treePath)
+        IndexDatabase indexDatabase = geogig.getRepository().indexDatabase();
+
+        if (indexDatabase.getIndex(tree, attribute).isPresent()) {
+            throw new CommandFailedException(
+                    "An index has already been created on that tree and attribute.");
+        }
+
+        Optional<RevFeatureType> featureTypeOpt = geogig.command(ResolveFeatureType.class)
+                .setRefSpec("HEAD:" + tree).call();
+        if (!featureTypeOpt.isPresent()) {
+            throw new CommandFailedException(
+                    String.format("Can't resolve '%s' as a feature type", tree));
+        }
+
+        RevFeatureType treeType = featureTypeOpt.get();
+
+        boolean attributeFound = false;
+        PropertyType attributeType = null;
+        for (PropertyDescriptor descriptor : treeType.descriptors()) {
+            if (descriptor.getName().toString().equals(attribute)) {
+                attributeFound = true;
+                attributeType = descriptor.getType();
+            }
```

Comment:
May want to `break;` out of the for loop here so it doesn't continue looping over the remaining attributes.